cardano-config/src/Cardano/Config/LedgerQueries.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE FlexibleInstances #-}
+
+module Cardano.Config.LedgerQueries
+  (LedgerQueries(..))
+where
+
+import           Prelude (Int, error, (.))
+
+import qualified Data.Map as Map
+
+import           Ouroboros.Consensus.Ledger.Abstract
+import           Byron.Spec.Ledger.Core (Relation(..))
+
+import qualified Cardano.Chain.Block as Byron
+import qualified Cardano.Chain.UTxO as Byron
+import qualified Ouroboros.Consensus.Byron.Ledger.Block as Byron
+import qualified Ouroboros.Consensus.Byron.Ledger.Ledger as Byron
+
+import qualified Ouroboros.Consensus.Shelley.Ledger as Shelley
+import qualified Shelley.Spec.Ledger.LedgerState as Shelley
+import qualified Shelley.Spec.Ledger.UTxO as Shelley
+
+import qualified Ouroboros.Consensus.Cardano as Cardano
+
+import qualified Ouroboros.Consensus.Mock.Ledger as Mock
+
+
+class LedgerQueries blk where
+  ledgerUtxoSize :: LedgerState blk -> Int
+
+instance LedgerQueries Byron.ByronBlock where
+  ledgerUtxoSize = size . Byron.unUTxO . Byron.cvsUtxo . Byron.byronLedgerState
+
+instance LedgerQueries (Shelley.ShelleyBlock c) where
+  ledgerUtxoSize =
+    (\(Shelley.UTxO xs)-> Map.size xs) . Shelley._utxo . Shelley._utxoState . Shelley.esLState . Shelley.nesEs . Shelley.shelleyState
+
+instance LedgerQueries (Cardano.CardanoBlock a) where
+  ledgerUtxoSize _ = error "ledgerUtxoSize:  not implemented for CardanoBlock"
```

Comment:
Easy to implement using the pattern synonyms from [`Ouroboros.Consensus.Cardano.Block`](https://github.com/input-output-hk/ouroboros-network/blob/f432deff2c9d94d8b8c67c7bb5c30b6cd6c579f7/ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Block.hs#L438):
```haskell
instance LedgerQueries (Cardano.CardanoBlock c) where
  ledgerUtxoSize = \case
    LedgerStateByron   ledgerByron   -> ledgerUtxoSize ledgerByron
    LedgerStateShelley ledgerShelley -> ledgerUTxoSize ledgerShelley
```

Reply:
Thank you, added.